### PR TITLE
Add highlight to current session backgroud

### LINKED ISSUE #521: [Assigned] High light current session card of the background
## Overview (Required)
- screenshot_20180129-112143

- I think that it is better to change the color of the background of session card so that we can more easily notice the currently open session

## Links
-

### PR DESCRIPTION
**Pull request by torotake**
## Issue
- close #521 

## Overview (Required)
- if current time is between Session.startDate and Session.endDate, set backgroud color to R.color.highlight_text.

## Links
- none

## Screenshot
screenshot_1517485188

### CODE REVIEW

**Comment by konifar** [comment]
Thanks! I think this highlight color is too vivid 🌈 
But we don't have a concrete design for this. Let me think about it for a while 🙇

**Comment by torotake** [comment]
Thank you for comment.
I feel this color is too vivid, too. 😆

I think that it may be better to change not only the color of the current session but also the background color of past sessions if there is an appropriate color.

**Comment by konifar** [comment]
Yea, exactly. I tried to create the draft design. If you like this, can you try to modify the layout? 🙏 

20180201_223231_nexus 5x

- `NEW!` => accent color
- `card background` => #fff9e6

**Comment by torotake** [comment]
Thanks. I think that your design is good!
I try to modify the layout according to this.

**Comment by torotake** [comment]
Implemented. How's this?

File: app/src/main/java/io/github/droidkaigi/confsched2018/presentation/sessions/item/SpecialSessionItem.kt [hunk 1]
```diff
@@ -15,6 +19,25 @@ data class SpecialSessionItem(
     override fun bind(viewBinding: ItemSpecialSessionBinding, position: Int) {
         viewBinding.session = session
         viewBinding.isShowDayNumber = isShowDayNumber
+
+        val now = Date()
+        when {
+            now.before(session.startTime) -> {
```
Comment (takahirom) [inline app/src/main/java/io/github/droidkaigi/confsched2018/presentation/sessions/item/SpecialSessionItem.kt]: Thanks! 
Can you add session.isOnGoing and use it?
And can you do it in xml with DataBinding?

**Comment by konifar** [comment]
👀

File: app/src/main/res/layout/item_special_session.xml [hunk 2]
```diff
@@ -36,17 +36,28 @@
...
+                android:layout_width="wrap_content"
+                android:layout_height="wrap_content"
+                android:layout_marginStart="8dp"
                 android:layout_marginTop="16dp"
                 android:text='@{"DAY"+session.dayNumber+" / "}'
                 android:visibility="@{isShowDayNumber ? View.VISIBLE : View.GONE}"
-                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintStart_toEndOf="@+id/current_badge"
                 app:layout_constraintTop_toTopOf="parent"
+                app:layout_goneMarginStart="12dp"
```
Comment (konifar) [inline app/src/main/res/layout/item_special_session.xml]: 👍

File: app/src/main/java/io/github/droidkaigi/confsched2018/presentation/sessions/item/SpecialSessionItem.kt (hunk 1 above)
Comment (torotake) [reply app/src/main/java/io/github/droidkaigi/confsched2018/presentation/sessions/item/SpecialSessionItem.kt]: Yeah, I challenge it!

**Comment by torotake** [comment]
I did it! It was good study about DataBinding for me this time.

File: model/src/main/java/io/github/droidkaigi/confsched2018/model/Session.kt [hunk 3]
```diff
@@ -37,4 +37,7 @@ sealed class Session(
 
     val isFinished: Boolean
         get() = System.currentTimeMillis() > endTime.time
+
+    val isOnGoing: Boolean
+        get() = Date().after(startTime) && Date().before(endTime)
```
Comment (takahirom) [inline model/src/main/java/io/github/droidkaigi/confsched2018/model/Session.kt]: Can you use a single instance for this? 🙇 💦 
For performance and safety.

```
System. currentTimeMillis().let { time -> startTime.time < time && time < endTime.time }
```

or

```
Date().let { date -> date.after(startTime) && date.before(endTime) }
```

**Comment by takahirom** [comment]
A lot thanks 👍 Amost LGTM 🥈

File: model/src/main/java/io/github/droidkaigi/confsched2018/model/Session.kt (hunk 3 above)
Comment (jmatsu) [reply model/src/main/java/io/github/droidkaigi/confsched2018/model/Session.kt]: I guess `System.currentTimeMillis() in startTime.time..endTime.time` is smart and can satisfy the concern @takahirom mentioned. 🤔

Comment (takahirom) [reply model/src/main/java/io/github/droidkaigi/confsched2018/model/Session.kt]: it's the best. 🥇

Comment (torotake) [reply model/src/main/java/io/github/droidkaigi/confsched2018/model/Session.kt]: Thank you for your advices. I fixed it following jmatsu's one.

**Comment by takahirom** [comment]
LGTM 🥇

**Pull request merged**